pkg/plugin/keyvault_test.go

```diff
@@ -133,16 +131,10 @@ func TestNewKeyVaultClient(t *testing.T) {
 		t.Run(test.desc, func(t *testing.T) {
 			kvClient, err := NewKeyVaultClient(test.config, test.vaultName, test.keyName, test.keyVersion, test.proxyMode, test.proxyAddress, test.proxyPort, test.managedHSM)
 			if err != nil {
-				t.Fatalf("newKeyVaultClient() failed with error: %v", err)
+				t.Fatalf("NewKeyVaultClient() failed with error: %v", err)
 			}
 			if kvClient == nil {
-				t.Fatalf("newKeyVaultClient() expected kv client to not be nil")
-			}
-			if !strings.Contains(kvClient.GetUserAgent(), "k8s-kms-keyvault") {
-				t.Fatalf("newKeyVaultClient() expected k8s-kms-keyvault user agent")
-			}
-			if kvClient.GetVaultURL() != test.expectedVaultURL {
```

Comment:
This is likely best as a unit test for `getVaultURL` and `getProxiedVaultURL`, speaks to a need for some refactor there to better encapsulate the logic rather than punching a hole through the public interface of the return from the constructor to test some implementation detail. WDYT?